### PR TITLE
CDPSDX-3557: Added new flow which is used for resize recovery in order to properly reconfigure the LB DNS for the old datalake in case of its deletion during recovery.

### PR DESCRIPTION
**Pull request by sxxgrc**
Jira: https://jira.cloudera.com/browse/CDPSDX-3557

Another new flow for all of the resize recovery fans.

The objective of this is to ensure that the following issue does not occur:

- Resize provisions a new medium duty DL with the same name and CRN as the original
- During provisioning, the LB DNS config in the PEM service is updated to match the new values
- For some reason provisioning then fails, or datalake restore fails, requiring the use of resize recovery
- Recovery deletes the new DL, in the process deleting the LB DNS config from the PEM service
- After recovery, while the old DL is up and running, none of the service endpoints can be accessed due to no entries for the DL's LB in the PEM service

So basically, I just added the update LB DNS step to the end of the resize recovery process. Note that checks are already in place for when this is not necessary.

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSActions.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
+                    StartUpdateLoadBalancerDNSEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, StartUpdateLoadBalancerDNSEvent payload,
+                    Map<Object, Object> variables) {
+                SdxCluster sdxCluster = sdxService.getById(payload.getResourceId());
```
Comment (bbihari) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSActions.java:67]: Could you please move the logic from here to a service (e.g. UpdateLoadBalancerDNSService).

File: datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSFlowConfig.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        implements RetryableFlowConfiguration<UpdateLoadBalancerDNSEvent> {
+    private static final List<Transition<UpdateLoadBalancerDNSState, UpdateLoadBalancerDNSEvent>> TRANSITIONS =
+        new Builder<UpdateLoadBalancerDNSState, UpdateLoadBalancerDNSEvent>()
+        .defaultFailureEvent(UPDATE_LOAD_BALANCER_DNS_FAILED_EVENT)
+        .from(INIT_STATE).to(UPDATE_LOAD_BALANCER_DNS_STATE)
+        .event(UPDATE_LOAD_BALANCER_DNS_EVENT).noFailureEvent()
+        .from(UPDATE_LOAD_BALANCER_DNS_STATE).to(FINAL_STATE)
+        .event(UPDATE_LOAD_BALANCER_DNS_SUCCESS_EVENT).defaultFailureEvent()
+        .from(UPDATE_LOAD_BALANCER_DNS_FAILED_STATE).to(FINAL_STATE)
+        .event(UPDATE_LOAD_BALANCER_DNS_FAILURE_HANDLED_EVENT).noFailureEvent()
```
Comment (kkalvagadda1) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSFlowConfig.java:31]: Please space each transition to make it readable.

**Review by kkalvagadda1 — COMMENTED** [review]
looks good to me.

**Review by RokolyA94 — APPROVED** [review]

**Review by bbihari — APPROVED** [review]

**Review by kkalvagadda1 — APPROVED** [review]

**Review by bergerdenes — APPROVED** [review]
